### PR TITLE
Refine dependency extraction: Ignore packages

### PR DESCRIPTION
**Pull request by allanrenucci**
We can ignore packages when collecting dependencies for incremental
compilation. We make sure not to ignore dependencies on package object.

### CODE REVIEW

**Comment by allanrenucci** [comment]
Zinc only ignores packages for class dependencies but not for used names. This PR also ignores packages as used names. Not sure it is safe

**Comment by allanrenucci** [comment]
@smarter What's your take on this? Should we add used name for packages like Zinc does?

**Comment by smarter** [comment]
I guess you need to because of package objects? If you import something from a package, and if the package object changes, then that may affect you

**Comment by allanrenucci** [comment]
I have added a test case that is failing whether or not we add used names for packages. The test also fails in Zinc (see sbt/zinc#530)

File: compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala [hunk 1]
```diff
@@ -295,8 +295,12 @@ private class ExtractDependenciesCollector extends tpd.TreeTraverser { thisTreeT
       val fromClass = resolveDependencySource
       if (fromClass.exists) { // can happen when visiting imports
         assert(fromClass.isClass)
-        _dependencies += ClassDependency(fromClass, enclOrModuleClass, DependencyByMemberRef)
+
         addUsedName(fromClass, mangledName(sym), UseScope.Default)
+        // packages have class symbol. Only record them as used names but bot dependency
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala]: typo: bot -> not

**Pull request merged**